node/src/blockchain/blockchain_bridge.rs

```diff
@@ -874,74 +983,198 @@ mod tests {
+                .unwrap()
+            })
+        );
+        assert_eq!(
+            processed_payments[1],
+            Correct(PendingPayable {
+                recipient_wallet: accounts_2.wallet,
+                hash: H256::from_str(
+                    "7f3221109e4f1de8ba1f7cd358aab340ecca872a1456cb1b4f59ca33d3e22ee3"
+                )
+                .unwrap()
             })
         );
+        let recording = accountant_recording.lock().unwrap();
+        assert_eq!(recording.len(), 1);
     }
 
     #[test]
-    fn blockchain_bridge_processes_requests_for_transaction_receipts_when_all_were_ok() {
-        let get_transaction_receipt_params_arc = Arc::new(Mutex::new(vec![]));
-        let (accountant, _, accountant_recording_arc) = make_recorder();
-        let pending_payable_fingerprint_1 = make_pending_payable_fingerprint();
-        let hash_1 = pending_payable_fingerprint_1.hash;
-        let hash_2 = make_tx_hash(78989);
-        let pending_payable_fingerprint_2 = PendingPayableFingerprint {
-            rowid: 456,
-            timestamp: SystemTime::now(),
-            hash: hash_2,
-            attempt: 3,
+    fn process_payments_fails_on_get_transaction_count() {
+        let test_name = "process_payments_fails_on_get_transaction_count";
+        let port = find_free_port();
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .response("trash transaction id".to_string(), 1)
+            .start();
+        let blockchain_interface_web3 = make_blockchain_interface_web3(Some(port));
+        let consuming_wallet = make_paying_wallet(b"consuming_wallet");
+        let system = System::new(test_name);
+        let agent = BlockchainAgentMock::default().consuming_wallet_result(consuming_wallet);
+        let msg = OutboundPaymentsInstructions::new(vec![], Box::new(agent), None);
+        let persistent_config = configure_default_persistent_config(ZERO);
+        let mut subject = BlockchainBridge::new(
+            Box::new(blockchain_interface_web3),
+            Box::new(persistent_config),
+            false,
+        );
+        let (accountant, _, accountant_recording) = make_recorder();
+        subject
+            .pending_payable_confirmation
+            .new_pp_fingerprints_sub_opt = Some(accountant.start().recipient());
+
+        let result = subject
+            .process_payments(msg.agent, msg.affordable_accounts)
+            .wait();
+
+        System::current().stop();
+        system.run();
+        let error_result = result.unwrap_err();
+        assert_eq!(
+            error_result,
+            TransactionID(BlockchainError::QueryFailed(
+                "Decoder error: Error(\"0x prefix is missing\", line: 0, column: 0) for wallet 0x2581…7849".to_string()
+            ))
+        );
+        let recording = accountant_recording.lock().unwrap();
+        assert_eq!(recording.len(), 0);
+    }
+
+    #[test]
+    fn process_payments_fails_on_missing_gas_price() {
```

Comment:
This test should go away as I explained in `blockchain_interface_web3/lower_level_interface.rs`. This belongs only and only into the BlockchainAgent, a few steps before  in the course of events...